### ISSUE TITLE
Helm chart is successfully deployed when using helm_repository datasource, but fails while providing repo URL directly to helm_release resource

### ISSUE THREAD

**Issue by mordowiciel** [hashicorp/terraform-provider-helm#561]
### Terraform Version and Provider Version

Terraform Version - 0.12.28
Helm Version - 3.2.4
Provider Version - 1.2.4

### Affected Resource(s)

- helm_release
- helm_repository

### Terraform Configuration Files
```hcl
resource "helm_release" "prometheus-yace-exporter" {
  chart = "prometheus-yace-exporter"
  repository = "https://<<redacted>>/repository/helm-hosted/"

  name = "prometheus-yace-exporter"
  namespace = "matching-store"

  create_namespace = true
}
```

### Debug Output

https://gist.github.com/mordowiciel/5803ff53999636e0d0a9d899f757b4b0

### Expected Behavior
Helm chart should be successfully deployed when providing URL repository directly in the ```helm_release``` resource.

### Actual Behavior
Helm Provider fails to download the chart from the provided repository URL:
```
helm_release.prometheus-yace-exporter: Creating...

Error: failed to download "https://<<redacted>>/repository/helm-hosted/prometheus-yace-exporter-1.0.0.tgz" (hint: running `helm repo update` may help)

  on main.tf line 7, in resource "helm_release" "prometheus-yace-exporter":
   7: resource "helm_release" "prometheus-yace-exporter" {
```
Hovever, I'm able to install the chart directly with Helm CLI, and, surprisingly, while using deprecated ```helm_repository``` data source:
```
data "helm_repository" "helm-hosted" {
  name = "helm-hosted"
  url  = "https://<<redacted>>/repository/helm-hosted/"
}


resource "helm_release" "prometheus-yace-exporter" {

  chart = "prometheus-yace-exporter"
  repository = data.helm_repository.helm-hosted.metadata[0].name

  name = "prometheus-yace-exporter"
  namespace = "matching-store"

  create_namespace = true
}
```

The following config also works when I manually add ```helm-hosted``` repository to Helm repo list before calling ```terraform apply```:
```
resource "helm_release" "prometheus-yace-exporter" {

  chart = "helm-hosted/prometheus-yace-exporter"

  name = "prometheus-yace-exporter"
  namespace = "matching-store"

  create_namespace = true
}
```

### Important Factoids

- The repository used in this example is private Helm repo, which is hosted on Sonatype Nexus Repository Manager OSS (v3.24.0-02)

### Steps to Reproduce

1. Create Helm repository on Sonatype Nexus Repository Manager OSS v3.24.0-02 and upload chosen test chart there.
2. Replace the repository URL with the URL of the created Nexus Helm repo.
3. ```terraform apply```

### References

None

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by alexsomesan** [comment]
related to #537

**Comment by jrhouston** [comment]
Thanks for opening this @mordowiciel, can you run your apply with `HELM_DEBUG=1` and see if you get a different error message?

**Comment by mordowiciel** [comment]
@jrhouston 

Hi, sorry for the late reply! Here's the result of the run with the debug flag:
https://gist.github.com/mordowiciel/4e64e7d26c4e37e44e285448f5fbb0a6

At this run, the interesting error had appeared:
`
2020/09/01 10:29:31 [ERROR] <root>: eval: *terraform.EvalApplyPost, err: no cached repo found. (try 'helm repo update'): open C:\Users\mordowiciel\AppData\Local\Temp\helm\repository\incubator-index.yaml: The system cannot find the file specified.
`

Any ideas what could cause this and how can I fix this?

**Comment by mordowiciel** [comment]
@jrhouston 

Hi, sorry for the late reply! Here's the result of the run with the debug flag:
https://gist.github.com/mordowiciel/4e64e7d26c4e37e44e285448f5fbb0a6

At this run, the interesting error had appeared:

`
2020/09/01 10:29:31 [ERROR] <root>: eval: *terraform.EvalApplyPost, err: no cached repo found. (try 'helm repo update'): open C:\Users\mordowiciel\AppData\Local\Temp\helm\repository\incubator-index.yaml: The system cannot find the file specified.
`

I've decided to remove all repositories I've added (including the incubator) and re-run the deployment, providing direct URL to repository to the helm_release resource (defined as in _Terraform Configuration Files_ section). This time it finished successfully.

Thanks for pointing out to the HELM_DEBUG flag! I'm closing this issue, as I'm considering it resolved. 

A little suggestion - it would be nice to forward this message even without the debug flag. This doesn't really seem as an error connected with downloading the chart from the repository.
